internal/controller/openstackimage_controller.go

```diff
@@ -147,16 +147,22 @@ func (r *OpenStackImageReconciler) reconcile(ctx context.Context, imageClient *g
 		logger.Info("resouce exists in OpenStack")
 	} else {
 		var err error
+
+		var imageVisibility images.ImageVisibility
+
+		if resource.Spec.Visibility != "default" {
+			imageVisibility = images.ImageVisibility(resource.Spec.Visibility)
+		}
```

Comment:
Here you instantiate `imageVisibility`, which instantly becomes the empty string. When the input is `"default"` the `if` doesn't apply; so we pass a pointer to an empty string to the API. Instead, we should instantiate a pointer and leave it `nil` if `default`.

I am well-aware that I wrote this error in your code... SORRY. The correct solution would be:

```suggestion
		var imageVisibility *images.ImageVisibility
		if resource.Spec.Visibility != "default" {
			v := images.ImageVisibility(resource.Spec.Visibility)
			imageVisibility = &v
		}
```

and below:
```
Visibility:      imageVisibility,
```